2.segment_images/scripts/3.reconstruct_3D_masks.py

```diff
@@ -304,20 +408,33 @@
 pos = nx.spring_layout(G)
 edge_labels = nx.get_edge_attributes(G, "weight")
 
+
+# In[ ]:
+
+
+# solve the graph to find the longest path with the shortest distance between nodes
+# also we need a constraint of about 10 um of total path length (a path length of 10 slices)
+# solve the graph to group the nodes into clusters based on the distance between them and their coordinates
+# longest_paths = nx.algorithms.community.greedy_modularity_communities(G)
```

Comment:
Cool approach! Consider adding a citation reference (or two) via a [`CITATION.cff`](https://citation-file-format.github.io/) file or similar (perhaps [referencing this](https://networkx.org/documentation/stable/index.html#citing)?).